### PR TITLE
Fix issues in assimp plugin and X files support

### PR DESCRIPTION
**Pull request by Meakk**
Implements https://github.com/f3d-app/f3d/issues/1124

### CODE REVIEW

File: plugins/assimp/module/vtkF3DAssimpImporter.cxx [hunk 1]
```diff
@@ -605,7 +605,7 @@ class vtkF3DAssimpImporter::vtkInternals
       }
 
       // convert materials to properties
-      this->Properties.resize(this->Scene->mNumMeshes);
+      this->Properties.resize(this->Scene->mNumMaterials);
```
Comment (Meakk) [inline plugins/assimp/module/vtkF3DAssimpImporter.cxx:608]: Fix the crash with `dwarf.x`

File: application/testing/CMakeLists.txt [hunk 2]
```diff
@@ -438,6 +438,7 @@ if(F3D_PLUGIN_BUILD_ASSIMP)
   f3d_test(NAME TestFBX DATA phong_cube.fbx ARGS --load-plugins=assimp DEFAULT_LIGHTS) # Using default lights because of ResetCamera
   f3d_test(NAME TestFBX16bits DATA 16bit.fbx ARGS --load-plugins=assimp THRESHOLD 70 DEFAULT_LIGHTS) # Using default lights because of ResetCamera
   f3d_test(NAME TestVerboseCameraAssimp DATA duck.dae ARGS --verbose --load-plugins=assimp NO_BASELINE REGEXP "camera1")
+  f3d_test(NAME TestX DATA anim_test.x ARGS --load-plugins=assimp DEFAULT_LIGHTS)
```
Comment (mwestphal) [inline application/testing/CMakeLists.txt:441]: update data licenses

Comment (Meakk) [reply application/testing/CMakeLists.txt:441]: already done

File: plugins/assimp/module/vtkF3DAssimpImporter.cxx (hunk 1 above)
Comment (mwestphal) [reply plugins/assimp/module/vtkF3DAssimpImporter.cxx:608]: do we need a test for this ?

Comment (Meakk) [reply plugins/assimp/module/vtkF3DAssimpImporter.cxx:608]: I don't think so. It was crashing when there was more materials than meshes. It's an edge case with weird datasets

File: resources/BundleInfo.plist.in [hunk 3]
```diff
@@ -326,6 +326,19 @@
...
+				<string>DirectX</string>
+				<key>CFBundleTypeRole</key>
+				<string>Viewer</string>
+				<key>LSIsAppleDefaultForType</key>
+				<true/>
+				<key>CFBundleTypeExtensions</key>
+				<array>
+					<string>x</string>
+				</array>
+			</dict>
```
Comment (mwestphal) [inline resources/BundleInfo.plist.in:341]: I completely forgot about this file, it misses many file formats and it should be configured depending on which plugins are enabled. Should I open an issue ?

Comment (mwestphal) [reply resources/BundleInfo.plist.in:341]: Ideally, it should be handled by the plugin framework instead of being its own thing.

**Review by mwestphal — APPROVED** [review]

File: resources/BundleInfo.plist.in (hunk 3 above)
Comment (Meakk) [reply resources/BundleInfo.plist.in:341]: Yeah you're right, it should be handled by plugins. Or maybe we shouldn't care about this file here and just handle it in the superbuild.  
Yes, please open an issue, we need to clarify that at some point.

Comment (mwestphal) [reply resources/BundleInfo.plist.in:341]: https://github.com/f3d-app/f3d/issues/1135
